frontend-react/src/pages/daily/Table/ReportsTable.test.tsx

```diff
@@ -1,72 +1,101 @@
-import * as SessionStorageTools from "../../../contexts/SessionStorageTools";
 import { historyServer } from "../../../__mocks__/HistoryMockServer";
+import * as ReportsHooks from "../../../hooks/network/History/ReportsHooks";
+import { TempRSReport } from "../../../network/api/History/Reports";
+import { mockSessionContext } from "../../../contexts/__mocks__/SessionContext";
+import {
+    MembershipController,
+    MemberType,
+} from "../../../hooks/UseOktaMemberships";
+import { SessionController } from "../../../hooks/UseSessionStorage";
 
-import ReportsTable from "./ReportsTable";
 import * as ReportUtilsModule from "./ReportsUtils";
+import ReportsTable from "./ReportsTable";
 
 const mockMs = (additional?: number) =>
     additional ? 1652458218417 + additional : 1652458218417;
-const mockAPIData: ReportResource[] = [
-    new ReportResource("1", mockMs(), mockMs(100000), 99, "CSV"),
-];
-const mockFetchReport = jest
-    .spyOn(ReportUtilsModule, "getReportAndDownload")
-    .mockImplementation(() => {
-        return mockAPIData[0];
-    });
-// @ts-ignore so we can mock this partially
-const mockAuth = jest.fn<() => Partial<IOktaContext>>().mockReturnValue({
-    authState: {
-        accessToken: {
-            accessToken: "",
-        },
-    },
-});
-jest.mock("@okta/okta-react", () => ({
-    useOktaAuth: () => mockAuth(),
-}));
-const mockStoredOrg = jest.spyOn(SessionStorageTools, "getStoredOrg");
-jest.mock("rest-hooks", () => ({
-    useResource: () => {
-        return mockAPIData;
-    },
-}));
+// const mockAPIData: ReportResource[] = [
+//     new ReportResource("1", mockMs(), mockMs(100000), 99, "CSV"),
+// ];
+const mockFetchReport = jest.spyOn(ReportUtilsModule, "getReportAndDownload");
+const mockApiHook = jest.spyOn(ReportsHooks, "useReportsList");
+const makeFakeData = (count: number) => {
+    const data: TempRSReport[] = [];
+    for (count; count > 0; count--) {
+        data.push({
+            reportId: `${count}`,
+            actions: [],
+            content: "",
+            displayName: "",
+            facilities: [],
+            fileName: "",
+            mimeType: "",
+            positive: 0,
+            receivingOrg: "",
+            receivingOrgSvc: "",
+            sendingOrg: "",
+            type: "",
+            via: "",
+            sent: mockMs(),
+            expires: mockMs(100000),
+            total: 99,
+            fileType: "CSV",
+        });
+    }
+    return data;
+};
 
 describe("ReportsTable", () => {
     beforeAll(() => historyServer.listen());
     afterEach(() => historyServer.resetHandlers());
     afterAll(() => historyServer.close());
-    beforeEach(() => renderWithRouter(<ReportsTable />));
-    test("renders with no error", () => {
-        // Hooks run
-        expect(mockAuth).toHaveBeenCalled();
-        expect(mockStoredOrg).toHaveBeenCalled();
-
+    beforeEach(() => {
+        mockSessionContext.mockReturnValue({
```

Comment:
Speaking of hand writing objects... a configurable mock objet generator for these would clean up just about every test file we have